Enhance localization and dynamic link generation

travlos opened the pull request:
Updated localized strings to fix typos and add support for dynamic placeholders in multiple languages. Enhanced `IdentityUILocalizer.cs` to enable dynamic link generation with parameters for URL, target, and CSS class. Updated `Profile.cshtml` and `Register.cshtml` to use these dynamic methods, replacing hardcoded links. Added new localized strings for additional labels and messages. Cleaned up `Terms.cs` by removing unused imports and redundant code.

These changes improve maintainability, flexibility, and the user experience, especially for multilingual users.

travlos committed:
Enhance localization and dynamic link generation

Updated localized strings to fix typos and add support for dynamic placeholders in multiple languages. Enhanced `IdentityUILocalizer.cs` to enable dynamic link generation with parameters for URL, target, and CSS class. Updated `Profile.cshtml` and `Register.cshtml` to use these dynamic methods, replacing hardcoded links. Added new localized strings for additional labels and messages. Cleaned up `Terms.cs` by removing unused imports and redundant code.

These changes improve maintainability, flexibility, and the user experience, especially for multilingual users.

travlos committed:
Standardize localization and simplify code structure

Updated localized strings to use fixed HTML attributes for
`target="_blank"` and `class="font-weight-700"` across all
locales (`el`, `es`, `it`, `ja`, and default`) for consistency.

Refactored `IdentityUILocalizer` methods to remove redundant
parameters (`target` and `cssClass`) and set default values
for `url`. Renamed methods for improved readability and clarity.

Updated `Register.cshtml` to align with refactored localization
methods, removing unnecessary parameters and ensuring proper
rendering of terms and privacy policy checkboxes.

These changes improve maintainability, readability, and ensure
a consistent user experience across the application.

travlos committed:
Refactor localization and URL handling in registration

Updated `IdentityUILocalizer` methods to include default URL
parameters and added remarks for customization via
`IdentityUIOptions`. Removed hardcoded `Url.PageLink` calls
in `Associate.cshtml` and `Register.cshtml`, relying on
default values instead. Improved consistency in checkbox
and label structures across views.

cleftheris commented on src/Indice.Features.Identity.UI/IdentityUILocalizer.cs:
```diff
@@ -935,12 +935,13 @@ public virtual HtmlString Register_Associate_your_Account(string provider) =>
     /// Instruction shown at the top of the registration form explaining what the user needs to do.
     /// </summary>
-    public virtual HtmlString Register_To_register_as_a_new_user_you_will_need_to_fill_in_the_following_information => new HtmlString(string.Format(CultureInfo.CurrentUICulture, IdentityLabels.Register_FillInInformation));
+    public virtual HtmlString Register_FillInInformation => new HtmlString(string.Format(CultureInfo.CurrentUICulture, IdentityLabels.Register_FillInInformation));
```
Rename to Register_Form_Instructions

travlos committed:
Renamed and reassigned resource keys and localized strings to improve clarity and consistency in the registration form.

- Renamed `Register_FillInInformation` to `Register_Form_Instructions` in `IdentityLabels.Designer.cs` and updated its usage across the application.
- Reassigned `Register_FirstName` to a different localized string for the "First name" field.
- Updated `.resx` files for all locales (`el`, `es`, `it`, `ja`, and default) to reflect the new `Register_Form_Instructions` key.
- Updated `IdentityUILocalizer.cs` to use the renamed `Register_Form_Instructions` property.
- Modified `Register.cshtml` to reference the updated `Register_Form_Instructions` key.

These changes enhance the maintainability and readability of the codebase by ensuring consistent naming and usage of localization keys.

cleftheris reviewed (approved):
🚀
